modelmatrix-core/src/main/scala/com/collective/modelmatrix/db/DatabaseConfig.scala

```diff
@@ -0,0 +1,52 @@
+package com.collective.modelmatrix.db
+
+import com.typesafe.config.{Config, ConfigFactory}
+import slick.driver.{H2Driver, JdbcProfile, PostgresDriver}
+
+
+// database configuration wrapper class that read the configuration files and
+// determines the proper slick driver to use
+class DatabaseConfigWrapper(configFilePath: String = "") {
+  // Differentiate between the different types of database supported
+  private [DatabaseConfigWrapper] case class DatabaseConfig(val name: String,
+                                                            val driverClass: String,
+                                                            val slickDriver: JdbcProfile)
+
+  private[this] val PG = DatabaseConfig("pg", "org.postgresql.Driver", PostgresDriver)
+  private[this] val H2 = DatabaseConfig("h2", "org.h2.Driver", H2Driver)
+
+  private[this] val dbConfigPath: String = "modelmatrix.catalog.db"
+
+  lazy val currentDB: DatabaseConfig = getCurrentDB
+  lazy val dbConfig: Config =
+    if (configFilePath.isEmpty) ConfigFactory.load().getConfig(dbConfigPath)
+    else ConfigFactory.systemProperties().withFallback(ConfigFactory.load(configFilePath)).getConfig(dbConfigPath)
+
```

Comment:
looks like currentDB and dbConfig are not used outside of wrapper class? in this case it can be private, and after that DatabaseConfig can be made private[this], because it will not leak outside of the Wrapper scope in currentDB method